ovs/src/main/java/org/dcsa/conformance/standards/ovs/checks/OvsChecks.java

```diff
@@ -0,0 +1,327 @@
+package org.dcsa.conformance.standards.ovs.checks;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import lombok.experimental.UtilityClass;
+import org.dcsa.conformance.core.check.ActionCheck;
+import org.dcsa.conformance.core.check.JsonAttribute;
+import org.dcsa.conformance.core.check.JsonContentCheck;
+import org.dcsa.conformance.core.traffic.HttpMessageType;
+import org.dcsa.conformance.standards.ovs.party.OvsFilterParameter;
+import org.dcsa.conformance.standards.ovs.party.OvsRole;
+import org.dcsa.conformance.standards.ovs.party.SuppliedScenarioParameters;
+
+import java.time.LocalDate;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+import java.util.function.BiPredicate;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+@UtilityClass
+public class OvsChecks {
+
+  public static ActionCheck responseContentChecks(
+      UUID matched, String standardVersion, Supplier<SuppliedScenarioParameters> sspSupplier) {
+
+    var checks = new ArrayList<JsonContentCheck>();
+
+    checks.add(
+        JsonAttribute.customValidator(
+            "Validate carrierServiceCode exists and "
+                + "matches in JSON response if request parameter has carrierServiceCode",
+            body ->
+                validateParameter(
+                    body,
+                    sspSupplier,
+                    OvsFilterParameter.CARRIER_SERVICE_CODE,
+                    "*/carrierServiceCode")));
+
+    checks.add(
+        JsonAttribute.customValidator(
+            "Validate universalServiceReference exists and "
+                + "matches in JSON response if request parameter has universalServiceReference",
+            body ->
+                validateParameter(
+                    body,
+                    sspSupplier,
+                    OvsFilterParameter.UNIVERSAL_SERVICE_REFERENCE,
+                    "*/universalServiceReference")));
+
+    checks.add(
+        JsonAttribute.customValidator(
+            "Validate vesselIMONumber exists and"
+                + " matches in JSON response if request parameter has vesselIMONumber",
+            body ->
+                validateParameter(
+                    body,
+                    sspSupplier,
+                    OvsFilterParameter.VESSEL_IMO_NUMBER,
+                    "*/vesselSchedules/*/vesselIMONumber")));
+
+    checks.add(
+        JsonAttribute.customValidator(
+            "Validate vesselName exists and"
```

Comment:
The checks are triggered only when the query parameter exists in the request and the json path for searched attribute exists in the response then the value is checked to make sure that publisher indeed provided the right data, otherwise the checks are passed. I hope this answers your question, else connect with me.

Reply:
Thanks Preetam for clarifying it.